src/platform/silabs/SiWx917/bluetooth/wfx_sl_ble_init.c

```diff
@@ -0,0 +1,539 @@
+
+    //  initializing the application events map
+    rsi_ble_app_init_events();
+
+    wfx_rsi.ble_task = xTaskCreateStatic((TaskFunction_t) rsi_ble_event_handling_task, "rsi_ble", WFX_RSI_TASK_SZ, NULL, 2,
+                                         wfxBLETaskStack, &rsiBLETaskStruct);
+    WFX_RSI_LOG("%s: rsi_task_suspend init_task ", __func__);
+    if (wfx_rsi.ble_task == NULL)
+    {
+        WFX_RSI_LOG("%s: error: failed to create ble task.", __func__);
+    }
+
+    WFX_RSI_LOG("%s complete", __func__);
+    //    rsi_task_destroy((rsi_task_handle_t *)wfx_rsi.init_task);
+    return RSI_SUCCESS;
+}
+
+/*==============================================*/
+/**
+ * @fn         rsi_ble_app_init_events
+ * @brief      initializes the event parameter.
+ * @param[in]  none.
+ * @return     none.
+ * @section description
+ * This function is used during BLE initialization.
+ */
+void rsi_ble_app_init_events()
+{
+    event_msg.ble_app_event_map  = 0;
+    event_msg.ble_app_event_mask = 0xFFFFFFFF;
+    event_msg.ble_app_event_mask = event_msg.ble_app_event_mask; // To suppress warning while compiling
+    WFX_RSI_LOG("Function :: rsi_ble_app_init_events");
+    return;
+}
+
+/*==============================================*/
+/**
+ * @fn         rsi_ble_app_clear_event
+ * @brief      clears the specific event.
+ * @param[in]  event_num, specific event number.
+ * @return     none.
+ * @section description
+ * This function is used to clear the specific event.
+ */
+void rsi_ble_app_clear_event(uint32_t event_num)
+{
+    event_msg.event_num = event_num;
+    event_msg.ble_app_event_map &= ~BIT(event_num);
+    return;
+}
+
+/*==============================================*/
+/**
+ * @fn         rsi_ble_on_mtu_event
+ * @brief      its invoked when mtu exhange event is received.
+ * @param[in]  rsi_ble_mtu, mtu event paramaters.
+ * @return     none.
+ * @section description
+ * This callback function is invoked when  mtu exhange event is received
+ */
+void rsi_ble_on_mtu_event(rsi_ble_event_mtu_t * rsi_ble_mtu)
+{
+    WFX_RSI_LOG(" RSI_BLE : rsi_ble_on_mtu_event");
+    memset(&event_msg.rsi_ble_mtu, 0, sizeof(rsi_ble_event_mtu_t));
+    memcpy(&event_msg.rsi_ble_mtu, rsi_ble_mtu, sizeof(rsi_ble_event_mtu_t));
+    rsi_ble_app_set_event(RSI_BLE_MTU_EVENT);
+}
+
+/*==============================================*/
+/**
+ * @fn         rsi_ble_on_gatt_write_event
+ * @brief      its invoked when write/notify/indication events are received.
+ * @param[in]  event_id, it indicates write/notification event id.
+ * @param[in]  rsi_ble_write, write event parameters.
+ * @return     none.
+ * @section description
+ * This callback function is invoked when write/notify/indication events are received
+ */
+void rsi_ble_on_gatt_write_event(uint16_t event_id, rsi_ble_event_write_t * rsi_ble_write)
+{
+    WFX_RSI_LOG(" RSI_BLE : rsi_ble_on_gatt_write_event");
+    memset(&event_msg.rsi_ble_write, 0, sizeof(rsi_ble_event_write_t));

```

Comment:
no need for this memset

Reply:
For BLE related, changes a new PR will be done. This doesn't support BLE commissioning. It will be handled in a new PR